kv.go

```diff
@@ -597,14 +604,30 @@ func (w *watcher) Updates() <-chan KeyValueEntry {
 	if w == nil {
 		return nil
 	}
+	w.mu.Lock()
+	defer w.mu.Unlock()
 	return w.updates
 }
 
+// close the update chan.
+func (w *watcher) close() {
+	if w == nil {
+		return
+	}
+	w.mu.Lock()
+	if w.updates != nil {
+		close(w.updates)
+		w.updates = nil
```

Comment:
As discussed in a call, the issue with that is that if user later calls w.Updates() then they would possibly range on a nil channel which would block.
That being said, maybe users could check that the channel is not nil after calling `Updates()`, but we discussed that you would want the channel to be closed, but not set to nil, so need to revisit that part.

Reply:
Yep will fix after a bit.

Reply:
ok changed, if you could take a look that would be great.